tests/mypy/modules/fail4.py

```diff
@@ -0,0 +1,20 @@
+"""
+Test mypy failure with dataclass.
+"""
+from typing import Optional
+
+from pydantic.dataclasses import dataclass
+
+
+class Config:
+    validate_assignment = True
+
+
+@dataclass(config=Config)
+class AddProject:
+    name: str
+    slug: Optional[str]
+    description: Optional[str]
+
+
+p = AddProject(name='x', slug='y', description='z')
```

Comment:
Is this failing because the plugin is not installed?

Please can we add a test to passes when the plugin is installed.

Reply:
Sorry, I fixed the test cases.